### PR TITLE
refactor(scan): use `better-sqlite3`

### PR DESCRIPTION
**Pull request by eventualbuddha**
## Overview

It has a cleaner API and is supposedly faster, though I don't think the speed difference will matter for our purposes and probably relies on prepared statements of which this commit mostly does not take advantage. Mostly I wanted it for the cleaner API, specifically around iterating over results rather than getting them all at once. The API is synchronous and already returns an iterator, giving me what I want already. In contrast, I had to wrap sqlite3's callback-based `each` with some gnarly manual promise stuff to make it work as an async iterator.
## Demo Video or Screenshot
n/a

## Testing Plan 
Automated testing plus manual testing of:
- [x] scan ballot
- [x] export CVRs
- [x] export backup

## Checklist
- [ ] ~I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.~
- [ ] ~I have added JSDoc comments to any newly introduced exports~

### CODE REVIEW

File: services/scan/src/db_client.test.ts [hunk 1]
```diff
@@ -2,141 +2,169 @@ import * as fs from 'fs-extra';
...
   expect(client.isMemoryDatabase()).toBe(false);
 
-  await client.exec(
+  client.exec(
     'create table if not exists muppets (name varchar(255) unique not null)'
   );
-  await client.run('insert into muppets (name) values (?)', 'Kermit');
-  await client.run('insert into muppets (name) values (?)', 'Fozzie');
+  client.run('insert into muppets (name) values (?)', 'Kermit');
+  client.run('insert into muppets (name) values (?)', 'Fozzie');
```
Comment (benadida) [inline services/scan/src/db_client.test.ts:19]: yay for synchronous muppets

File: services/scan/src/server.test.ts [hunk 2]
```diff
@@ -140,28 +140,28 @@ test('GET /config/election (application/json)', async () => {
...
     status: 'ok',
     testMode: true,
   });
 });
 
-test('GET /config/markThresholdOverrrides', async () => {
-  await workspace.store.setElection(testElectionDefinition);
-  await workspace.store.setTestMode(true);
-  await workspace.store.setMarkThresholdOverrides({
+test('GET /config/markThresholdOverrides', async () => {
```
Comment (benadida) [inline services/scan/src/server.test.ts:161]: I had to stare at this one for a while to figure out what changed :)

**Review by benadida — APPROVED** [review]
so much easier to read!

Looks good but obviously needs to be seriously tested on real hardware etc.

**Comment by eventualbuddha** [comment]
Verified this all works on a Lenovo IdeaPad Flex.

**Pull request merged**